src/main/java/competition/subsystems/oracle/OracleSubsystem.java

```diff
@@ -117,13 +118,36 @@ public List<XbotSwervePoint> getRecommendedScoringTrajectory() {
     }
 
     public List<XbotSwervePoint> getRecommendedCoralPickupTrajectory() {
-        // TODO: go to more than one location.
-        var finalWaypoint = Landmarks.BlueLeftCoralStationMid;
-        var route = blueReefRoutingCircle.generateSwervePoints(pose.getCurrentPose2d(), finalWaypoint);
+        var goalCoralStation = getCoralStation(pose.getCurrentPose2d());
+        var route = blueReefRoutingCircle.generateSwervePoints(pose.getCurrentPose2d(), goalCoralStation);
         aKitLog.record("RecommendedRoute", XbotSwervePoint.generateTrajectory(route));
         return route;
     }
 
+    private Pose2d getCoralStation(Pose2d currentPose) {
+        Pose2d leftStation = PoseSubsystem.convertBlueToRedIfNeeded(Landmarks.BlueLeftCoralStationMid);
+        Pose2d rightStation = PoseSubsystem.convertBlueToRedIfNeeded(Landmarks.BlueRightCoralStationMid);
+        double leftStationDistance = currentPose.getTranslation().getDistance(leftStation.getTranslation());
+        double rightStationDistance = currentPose.getTranslation().getDistance(rightStation.getTranslation());
+
+        // See if we have constraints set
+        return switch (allowedCoralStations) {
+            case ONLY_LEFT_STATION -> leftStation;
+            case ONLY_RIGHT_STATION -> rightStation;
+            case NO_STATION -> currentPose;
+            case ANY_STATION -> leftStationDistance < rightStationDistance ? leftStation : rightStation;
+        };
+    }
+
+    public void setAllowedCoralStations(CoralStationMode mode) {
```

Comment:
⭐ I think the name of this method is a little different from what it's doing, consider renaming
```suggestion
    public void setCoralStationMode(CoralStationMode mode) {